bin/echo.c

```diff
@@ -165,10 +215,15 @@ int echo(struct s2n_connection *conn, int sockfd)
     errno = 0;
 
     /* Act as a simple proxy between stdin and the SSL connection */
-    int p;
+    int p = 0;
     s2n_blocked_status blocked;
     do {
-        while ((p = poll(readers, 2, -1)) > 0) {
+        /* echo will send and receive Application Data back and forth between
+         * client and server, until stop_echo is true. */
+        while (!(*stop_echo)) {
+            if ((p = poll(readers, 2, -1)) < 0) {
+                return 0;
```

Comment:
if p < 0, there was an error, we should print the error to stderr and exit echo instead of returning success. 

```
fprintf(stderr, "Poll error: %s\n", strerror(errno));
exit(1);
```

Reply:
@ttjsu-aws Let's not mess with existing echo error handling as part of this change. It's not required.

However, you are accidentally changing existing error handling behavior with the current refactor. That sneaky outer `do` actually catches and retries one particular error, but you're now bypassing that. That's why I suggested just breaking for *stop_echo instead. It'll still ultimately handle <0 by returning 0, but it'll let the outer loop retry EINTR.

Reply:
I am not suggesting changing the existing error handling here. `echo` currently `returns 0` only for success and for failure cases in `echo` we dont return and instead exit.

Reply:
We actually succeed for poll < 0 (if the outer loop doesn't catch and retry), so exiting with an error there would change behavior.